### PR TITLE
api(context): get rid of PageEvent

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: test/browsercontext.spec.js [hunk 1]
```diff
@@ -547,12 +551,12 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, FF
     });
     it('should report initialized pages', async({browser, server}) => {
       const context = await browser.newContext();
-      const pagePromise = context.waitForEvent('page').then(e => e.page());
+      const pagePromise = context.waitForEvent('page');
```
Comment (dgozman) [inline test/browsercontext.spec.js]: Let's add a test for `page.waitForLoadState({waitUntil:'load'})` on a `context.newPage()`, about:blank popup and network url popup. We lost the coverage by removing `e.page()` calls that were waiting for load.

File: test/navigation.spec.js [hunk 2]
```diff
@@ -835,10 +835,11 @@ module.exports.describe = function({testRunner, expect, playwright, MAC, WIN, FF
     it('should work with pages that have loaded before being connected to', async({page, context, server}) => {
       await page.goto(server.EMPTY_PAGE);
       const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page()),
+        page.waitForEvent('popup'),
         page.evaluate(() => window._popup = window.open(document.location.href)),
       ]);
-      expect(popup.url()).toBe(server.EMPTY_PAGE);
+      // The url is about:blank in FF.
+      // expect(popup.url()).toBe(server.EMPTY_PAGE);
```
Comment (dgozman) [inline test/navigation.spec.js]: Let's mark it as failed, or we'll forget to fix it.

File: test/popup.spec.js [hunk 3]
```diff
@@ -205,73 +206,64 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
           const win = window.open('about:blank');
           win.close();
         }),
       ]);
-      expect(popupEvent).toBeTruthy();
+      expect(popup).toBeTruthy();
       await context.close();
     });
-    it('should resolve page() after initial navigation', async({browser, server}) => {
+    it('should resolve page after initial navigation', async({browser, server}) => {
```
Comment (dgozman) [inline test/popup.spec.js]: We can just remove this test.

File: test/popup.spec.js [hunk 4]
```diff
@@ -205,73 +206,64 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
       expect(resolved).toBe(false);
       cssResponse.end('');
-      const popup = await popupPromise;
+      await loadSatePromise;
       expect(resolved).toBe(true);
       expect(popup.url()).toBe(server.PREFIX + '/one-style.html');
       await context.close();
     });
-    it('should respect timeout in page()', async({browser, server}) => {
+    it('should respect timeout in waitForLoadState()', async({browser, server}) => {
```
Comment (dgozman) [inline test/popup.spec.js]: Let's remove this test.

File: test/popup.spec.js [hunk 5]
```diff
@@ -284,52 +276,29 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
-      const popup = await popupEvent.page();
+      const popup = await page.waitForEvent('popup');
       const dialog = await popup.waitForEvent('dialog');
       expect(dialog.message()).toBe('hello');
       await dialog.dismiss();
       await evaluatePromise;
       await context.close();
     });
-    it.fail(FFOX)('should work with empty url', async({browser}) => {
+    it('should work with empty url', async({browser}) => {
```
Comment (dgozman) [inline test/popup.spec.js]: Let's keep a test which opens an empty url, and then calls `waitForLoadState({waitUntil:'load'})`.

File: test/popup.spec.js [hunk 6]
```diff
@@ -284,52 +276,29 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
-      const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page()),
-        page.evaluate(() => window.__popup = window.open('about:blank', null, 'noopener')),
-      ]);
-      expect(await page.evaluate(() => !!window.opener)).toBe(false);
-      expect(await popup.evaluate(() => !!window.opener)).toBe(false);
-      await context.close();
-    });
-    it('should work with noopener and nowait', async({browser}) => {
+    it('should work with noopener and about:blank', async({browser}) => {
```
Comment (dgozman) [inline test/popup.spec.js]: Same here - we had two tests to check that we can resolve early, and also can wait for load. Let's have two?

File: test/popup.spec.js [hunk 7]
```diff
@@ -341,19 +310,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
-      expect(await popup.evaluate(() => !!window.opener)).toBe(false);
-      await context.close();
-    });
-    it('should work with noopener and url and nowait', async({browser, server}) => {
-      const context = await browser.newContext();
-      const page = await context.newPage();
-      await page.goto(server.EMPTY_PAGE);
-      const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page({ waitUntil: 'nowait' })),
+        page.waitForEvent('popup'),
```
Comment (dgozman) [inline test/popup.spec.js]: ditto

File: test/popup.spec.js [hunk 8]
```diff
@@ -366,20 +323,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
...
-      await context.close();
-    });
-    it('should work with clicking target=_blank and nowait', async({browser, server}) => {
-      const context = await browser.newContext();
-      const page = await context.newPage();
-      await page.goto(server.EMPTY_PAGE);
-      await page.setContent('<a target=_blank rel="opener" href="/one-style.html">yo</a>');
-      const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page({ waitUntil: 'nowait' })),
+        page.waitForEvent('popup'),
```
Comment (dgozman) [inline test/popup.spec.js]: ditto

**Review by dgozman — APPROVED** [review]

File: test/popup.spec.js (hunk 4 above)
Comment (yury-s) [reply test/popup.spec.js]: Done.

File: docs/api.md [hunk 9]
```diff
@@ -312,17 +311,16 @@ Emitted when Browser context gets closed. This might happen because of one of th
 - The [`browser.close`](#browserclose) method was called.
 
 #### event: 'page'
-- <[PageEvent]>
+- <[Page]> 
 
-Emitted when a new Page is created in the BrowserContext. The event will also fire for popup
-pages. See also [`Page.on('popup')`](#event-popup) to receive events about popups relevant to a specific page.
+The event is emitted when a new Page is created in the BrowserContext. The page may still be navigating. Use [Page.waitForLoadState](#pagewaitforloadstateoptions) to wait until the page gets to a particular state. The event will also fire for popup pages. See also [`Page.on('popup')`](#event-popup) to receive events about popups relevant to a specific page.
```
[the remark this answers is not in the document]
Comment (yury-s) [reply docs/api.md]: Done.

File: docs/api.md [hunk 10]
```diff
@@ -1815,34 +1813,6 @@ This method returns all of the dedicated [WebWorkers](https://developer.mozilla.
...
-    - `'domcontentloaded'` - consider navigation to be finished when the `DOMContentLoaded` event is fired.
-    - `'load'` - consider navigation to be finished when the `load` event is fired.
-    - `'networkidle0'` - consider navigation to be finished when there are no more than 0 network connections for at least `500` ms.
-    - `'networkidle2'` - consider navigation to be finished when there are no more than 2 network connections for at least `500` ms.
-- returns: <[Promise]<[Page]>> Promise which resolves to the created page once it loads, according to `waitUntil` option.
-
-This resolves when the page reaches a required load state, `load` by default. The earliest moment that page is available is when it has navigated to the initial url (corresponds to `{waitUntil: 'nowait'}` option). For example, when opening a popup with `window.open('http://example.com')`, this method will wait until the network request to "http://example.com" is done and its response has started loading in the popup. Passing different `waitUntil` options will also wait for the particular load state to happen, e.g. default `load` waits until the load event fires.
```
[the remark this answers is not in the document]
Comment (yury-s) [reply docs/api.md]: Done.

File: src/chromium/crBrowser.ts [hunk 11]
```diff
@@ -130,19 +130,21 @@ export class CRBrowser extends platform.EventEmitter implements Browser {
...
+      const page = target._crPage!.page();
       if (targetInfo.type === 'page') {
         this._firstPageCallback();
-        context.emit(CommonEvents.BrowserContext.Page, pageEvent);
+        context.emit(CommonEvents.BrowserContext.Page, page);
         const opener = target.opener();
         if (!opener)
           return;
+        // Opener page must have been initialized already and resumed in order to
+        // create this popup, maybe use opener._crPage!.page()?
```
[the remark this answers is not in the document]
Comment (yury-s) [reply src/chromium/crBrowser.ts]: On the second thought I believe there is a slight chance of race: the opener page may have been fully initialized (and resumed) in the browser but playwright didn't receive all responses yet, child page may be created and initialized before the response comes.

File: test/browsercontext.spec.js [hunk 12]
```diff
@@ -39,7 +39,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, FF
       const page = await context.newPage();
       await page.goto(server.EMPTY_PAGE);
       const [popup] = await Promise.all([
-        utils.waitEvent(page, 'popup').then(e => e.page()),
+        utils.waitEvent(page, 'popup'),
```
[the remark this answers is not in the document]
Comment (yury-s) [reply test/browsercontext.spec.js]: Done.

File: test/navigation.spec.js (hunk 2 above)
Comment (yury-s) [reply test/navigation.spec.js]: Done.

File: test/popup.spec.js (hunk 3 above)
Comment (yury-s) [reply test/popup.spec.js]: Done.

File: test/popup.spec.js (hunk 5 above)
Comment (yury-s) [reply test/popup.spec.js]: Done, added to Page.waitForLoadState suite

File: docs/api.md [hunk 13]
```diff
@@ -765,7 +763,7 @@ Emitted when the JavaScript [`load`](https://developer.mozilla.org/en-US/docs/We
 Emitted when an uncaught exception happens within the page.
 
 #### event: 'popup'
-- <[PageEvent]> Page event corresponding to "popup" window
+- <[Page]> Page corresponding to "popup" window
 
 Emitted when the page opens a new tab or window. This event is emitted in addition to the [`browserContext.on('page')`](#event-page), but only for popups relevant to this page.
```
[the remark this answers is not in the document]
Comment (yury-s) [reply docs/api.md]: Fixed.

**Pull request merged**